model/src/main/kotlin/licenses/LicenseInfoResolver.kt

```diff
@@ -131,15 +131,23 @@ class LicenseInfoResolver(
                 findings.licenses,
                 findings.licenseFindingCurations,
                 findings.relativeFindingsPath
-            ).mapNotNull { it.curatedFinding?.license }
-        }
+            ).mapNotNull { curationResult ->
+                val licenseFinding = curationResult.curatedFinding ?: return@mapNotNull null
+
+                licenseFinding.license to findings.pathExcludes.any { pathExclude ->
+                    pathExclude.matches(licenseFinding.location.prependPath(findings.relativeFindingsPath))
+                }
+            }
+        }.groupBy(keySelector = { it.first }, valueTransform = { it.second }).mapValues { it.value.all { it } }
 
         resolvedLocations.keys.forEach { license ->
             license.builder().apply {
                 resolvedLocations[license]?.let { locations.addAll(it) }
 
-                originalExpressions += detectedLicenses.filterTo(mutableSetOf()) { license in it.decompose() }.map {
-                    ResolvedOriginalExpression(expression = it, source = LicenseSource.DETECTED)
+                originalExpressions += detectedLicenses.entries.filter { (spdxExpression, _) ->
+                    license in spdxExpression.decompose()
+                }.map { (expression, isDetectedExcluded) ->
```

Comment:
For consistency either rename `spdxExpression` to `expression` in the `filter` call or `expression` to `spdxExpression` here.